rust/read_file.rs

```diff
@@ -0,0 +1,21 @@
+use std::env;
+use std::fs::File;
+use std::io::{BufReader, BufRead};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = env::args().collect();
+    let file_path = &args[1];
+
+    let file = File::open(file_path)?;
+    let reader = BufReader::new(file);
+
+    let mut line_number = 1;
+
+    for line_result in reader.lines() {
```

Comment:
```suggestion
    for (line_number, line_result) in reader.lines().enumerate() {
```
We can do this to avoid needing to initialize and implement `line_number` ourselves.